src/controllers/profile.js

```diff
@@ -0,0 +1,68 @@
+const ErrorResponse = require('../utils/errorResponse');
+const Ngo = require('../models/ngo');
+const Volunteer = require('../models/volunteer');
+
+exports.getProfile = async (req, res, next) => {
+  if (req.user) {
+    const user =
+      (await Volunteer.findOne(
+        // eslint-disable-next-line no-underscore-dangle
+        { _id: req.user._id },
+        { password: 0, createdAt: 0, updatedAt: 0, _id: 0, __v: 0 }
+      )) ||
+      (await Ngo.findOne(
+        // eslint-disable-next-line no-underscore-dangle
+        { _id: req.user._id },
+        { password: 0, createdAt: 0, updatedAt: 0, _id: 0, __v: 0 }
+      ));
+    return res.status(200).json({
+      success: true,
+      data: user,
+    });
+  }
+  return 'You are not logged in';
+};
+
+exports.updateProfile = async (req, res, next) => {
+  if (req.user) {
+    const user =
+      (await Volunteer.findOneAndUpdate(
```

Comment:
Before we update, let's make sure that the new email/username (`req.body.email`), are not used by other ngos/volunteers.
just like you checked it at signup